src/serac/physics/solid_functional.hpp

```diff
@@ -405,11 +441,16 @@ class SolidFunctional<order, dim, Parameters<parameter_space...>, std::integer_s
   void addBodyForce(BodyForceType body_force)
   {
     residual_->AddDomainIntegral(
-        Dimension<dim>{}, DependsOn<0, 1>{},
-        [body_force, this](auto x, auto /* displacement */, auto /* acceleration */, auto... /*params*/) {
+        Dimension<dim>{}, DependsOn<0, 1, 2>{},
+        [body_force, this](auto x, auto /* displacement */, auto /* acceleration */, auto shape, auto... /* params */) {
+          auto dp_dX = get<DERIVATIVE>(shape);
+          // auto p     = get<VALUE>(shape);
+
           // note: this assumes that the body force function is defined
           // per unit volume in the reference configuration
-          return serac::tuple{body_force(x, time_), zero{}};
+
+          // TODO should we add the shape parameter to x?
+          return serac::tuple{body_force(x, time_) * det(dp_dX + I), zero{}};
```

Comment:
> TODO should we add the shape parameter to x?

I think so, and maybe we should be adding the displacement as well. We should decide which configuration the physical coordinates correspond to for the body force terms.

Reply:
+1 for adding the shape displacement to the coordinates once and for all. 

additional point: I vote to keep the reference coordinates as the default argument for the loading functions. Dead loading is both common and simpler (since it doesn't entail geometric nonlinearity). We *could* make an overload for a spatial version, but I suggest waiting until a user request for it comes in.